front/app/components/FilterSelector/MultiSelectDropdown.tsx

```diff
@@ -122,38 +122,52 @@ const MultiSelectDropdown = ({
   isLoading,
 }: Props) => {
   const tabsRef = useRef<(HTMLLIElement | null)[]>([]);
-  const [focusedIndex, setFocusedIndex] = useState(-1);
   const isPhoneOrSmaller = useBreakpoint('phone');
 
   const handleOnToggleCheckbox =
     (entry: IFilterSelectorValue) => (_event: React.ChangeEvent) => {
       onChange(entry.value);
     };
 
-  const handleOnSelectSingleValue =
-    (entry: IFilterSelectorValue) => (event: KeyboardEvent<HTMLLIElement>) => {
-      if (
-        event.type === 'keydown' &&
-        (event.key === 'ArrowUp' || event.key === 'ArrowDown')
-      ) {
-        event.preventDefault();
-        const totalItems = values.length;
-        let nextIndex = 0;
-        if (event.key === 'ArrowUp') {
-          nextIndex = focusedIndex === 0 ? totalItems - 1 : focusedIndex - 1;
-        } else {
-          nextIndex = focusedIndex === totalItems - 1 ? 0 : focusedIndex + 1;
+  const handleOnKeyDown =
```

Comment:
I know you didn't introduce this but this method needs to be cleaned up. Don't call a function that calls a function, collapse all arguments into a single function call.